internal/util/util.go

```diff
@@ -49,7 +49,7 @@ func MapBundleToBundleInstanceHandler(cl client.Client, log logr.Logger) handler
 		}
 		for _, bi := range bundleInstances.Items {
 			bi := bi
-			if bi.Spec.BundleName == b.Name {
+			if bi.Spec.Template.Name == b.Name {
```

Comment:
TODO: this seems like the wrong comparison in the context of the current implementation. We'd need to either parse the current Bundle's owner references, or compare the Bundle's label selectors with the current BI's spec.Selector values.